python/cudf/cudf/io/parquet.py

```diff
@@ -120,12 +124,18 @@ def write_to_dataset(
                         write_df.to_parquet(
                             fil,
                             index=preserve_index,
+                            row_group_cols=row_group_cols,
                             metadata_file_path=fs.sep.join([subdir, filename]),
                             **kwargs,
                         )
                     )
                 else:
-                    write_df.to_parquet(fil, index=preserve_index, **kwargs)
+                    write_df.to_parquet(
+                        fil,
+                        index=preserve_index,
+                        row_group_cols=row_group_cols,
+                        **kwargs,
+                    )
```

Comment:
I think this section of code can be written to only call `to_parquet` once. Something roughly like this, which updates `**kwargs` and optionally keeps the result:

```python
if return_metadata:
    kwargs["metadata_file_path"] = fs.sep.join([subdir, filename])

metadata_result = write_df.to_parquet(
    fil,
    index=preserve_index,
    row_group_cols=row_group_cols,
    **kwargs,
)

if return_metadata:
    metadata.append(metadata_result)
```